app/assets/javascripts/govuk_publishing_components/analytics-ga4/ga4-enhanced-ecommerce-tracker.js

```diff
@@ -0,0 +1,133 @@
+window.GOVUK = window.GOVUK || {}
+window.GOVUK.Modules = window.GOVUK.Modules || {};
+
+(function (Modules) {
+  'use strict'
+
+  function Ga4EnhancedEcommerceTracker (module) {
+    this.module = module
+  }
+
+  Ga4EnhancedEcommerceTracker.prototype.init = function (isNewPage = true) {
```

Comment:
We don't normally pass parameters to the `init` function of a module like this, normally because Modules are called and initialised by the start Modules script. Instead, normal practice is to have another data attribute on the element in question and read it when the constructor is called, or in the init function. For example:

```HTML
<div data-module="Ga4EnhancedEcommerceTracker" data-is-new-page="false"></div>
```

```JavaScript
  function Ga4EnhancedEcommerceTracker (module) {
    this.module = module
    this.isNewPage = this.module.getAttribute('data-is-new-page') || true
  }
```